Attack.py

```
@@ -102,1366 +86,676 @@ def __call__(self, *args: Any, **kwargs: Any) -> "_NXStub":
+        logger.info(f"Generating {n} synthetic samples...")
+        X, y = [], []
+        for _ in range(n // 2):
+            row = [secrets.randbelow(1 << 120) for _ in range(6)]
+            X.append(self._features(self._good(row)))
+            y.append(1)
+            X.append(self._features(self._bad(row)))
+            y.append(0)
+        X = [x for x in X if x is not None]
+        y = y[:len(X)]
+        if not X:
+            logger.warning("No valid synthetic features – ML disabled")
```

Ruff (0.14.2)

String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)